### PR TITLE
Redshift.clj to db.clj

### PR DESCRIPTION
**Pull request by coconutpalm**
Ready for feedback/thumbs.  😄 

Moving BEETL's redshift.clj to clj-foundation/.../db.clj so Levi can consume it without depending on BEETL.

Approach:
- Rename redshift.clj -> db.clj
- Remove code from db.clj that was duplicated in templates.clj
- Move error timeout/retry logic to errors.clj
- Support a default database but don't hard-code redshift as The One
- Add tests for any code I change/touch that wasn't already tested

### CODE REVIEW

**Comment by coconutpalm** [comment]
The timeout / retry logic from the DB job queue (but not the queue itself) has been migrated to errors.clj and tested in errors-test.clj.

Feedback appreciated.

File: project.clj [hunk 1]
```diff
@@ -15,4 +15,6 @@
   :dependencies [[org.clojure/clojure "1.8.0"]
                  [org.clojure/tools.logging "0.3.1"]
                  [prismatic/schema "1.1.1"]
-                 [potemkin "0.4.3"]])
+                 [potemkin "0.4.3"]
+                 [org.clojure/java.jdbc "0.6.2-alpha2"]
+                 [com.h2database/h2 "1.4.192"]])
```
Comment (coconutpalm) [inline project.clj]: Move to test profile

**Comment by baguthrie** [comment]
👍

File: src/clj_foundation/db.clj [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+
+  * JOB-NAME   (defaults to the first line of SQL if not specified)
+  * ABORT?-FN  (defaults to clj-foundation.db/any-fatal-exceptions?)
+
+  Refer to the docstrings for the above defs for more information on each setting.
+
+  Returns a function that can accept additional kv parameters and execute the PreparedStatement.
+
+  That function returns the results of calling:
+  (sql-fn connection [prepared-statement] bind-arg1 bind-arg2 ... bind-argn {dblib-params})"
```
Comment (coconutpalm) [inline src/clj_foundation/db.clj]: Fix syntax of calling sql-fn

**Comment by gsavovski** [comment]
👍

**Pull request merged**